samples/net/connectivity_check/README.rst

```diff
@@ -0,0 +1,146 @@
+.. zephyr:code-sample:: online-connectivity-check
+   :name: Online Connectivity Checker for Connection Manager
+   :relevant-api: bsd_sockets http_client tls_credentials secure_sockets_options
+
+   Use online connectivity checker in connection manager to verify that
+   we have connectivity to a user-specified network/host.
+
+Overview
+********
+
+This sample application enables the online connectivity checker in network
+connection manager. The checker tries to connect to user-specified host either
+by an ICMP ping request, or by a HTTP(S) GET request.
+If the connection can be established, a ``NET_EVENT_CONNECTIVITY_ONLINE`` is
+generated and application can monitor this event and then act on it.
+If the connectivity to user-specified host is lost, then
+a ``NET_EVENT_CONNECTIVITY_OFFLINE`` event is generated.
+
+The source code for this sample application can be found at:
+:zephyr_file:`samples/net/connectivity_check`.
+
+Requirements
+************
+
+- :ref:`networking_with_host`
+
+Building and Running
+********************
+
+You can use this application on a supported board, including
+running it inside QEMU as described in :ref:`networking_with_qemu`.
+
+Build the http-client sample application like this:
+
+.. zephyr-app-commands::
+   :zephyr-app: samples/net/connectivity_check
+   :board: <board to use>
```

Comment:
Hmm, I don't think the instructions are very qemu centric. I mention qemu once in the readme file. Perhaps we could call this emulator centric instructions as also native_sim board is mentioned.

Reply:
I added more information if user wants to use some other board to run the sample.

Reply:
Ya, maybe emulator centric then :) My concern is with the "As the connectivity check is meant to connect to some outside network with public IP addresses, it might be necessary to setup network settings on the host side so that Linux host can act as a router." paragraph -- when I first read it I was literally like "wait, what, it said earlier I can run this sample on a supported board, so what is this referring to?"

Maybe turn the two following paragraphs into a `note` aimed at people running in an emulated environment?

Reply:
I changed the chapter to `Running connectivity_check against Docker Container`, would that be enough here? Also added information that these instructions are meant for testing purposes.

Reply:
please push your changes tho :)